Source/Core/Common/MemArena.h

```diff
@@ -15,18 +15,57 @@ namespace Common
 {
 // This class lets you create a block of anonymous RAM, and then arbitrarily map views into it.
 // Multiple views can mirror the same section of the block, which makes it very convenient for
-// emulating
-// memory mirrors.
-class MemArena
+// emulating memory mirrors.
+class MemArena final
 {
 public:
+  MemArena();
+  ~MemArena();
+  MemArena(const MemArena&) = delete;
+  MemArena(MemArena&&) = delete;
+  MemArena& operator=(const MemArena&) = delete;
+  MemArena& operator=(MemArena&&) = delete;
+
+  // Allocate a memory segment that can be then interacted with using CreateView() and
+  // ReleaseView(). This is used to make a mappable region for emulated memory.
+  // 'size' is the amount of bytes available in this region.
   void GrabSHMSegment(size_t size);
+
+  // Release the memory segment previously allocated with GrabSHMSegment().
   void ReleaseSHMSegment();
-  void* CreateView(s64 offset, size_t size, void* base = nullptr);
+
+  // Map a memory region in the memory segment previously allocated with GrabSHMSegment().
+  // 'offset': Offset within the memory segment to map at.
+  // 'size': Size of the region to map.
+  // Returns a pointer to the memory region, or nullptr on failure.
+  void* CreateView(s64 offset, size_t size);
+
+  // Unmap a memory region previously mapped with CreateView().
+  // 'view': Pointer returned by CreateView().
+  // 'size': Size passed to the corresponding CreateView() call.
   void ReleaseView(void* view, size_t size);
 
-  // This finds 1 GB in 32-bit, 16 GB in 64-bit.
-  static u8* FindMemoryBase();
+  // Reserve a memory region, but don't allocate anything within it.
+  // This is used to create our 'fastmem' memory area for the emulated game code to access directly.
+  // 'memory_size': Size in bytes of the memory region to reserve.
+  // Returns a pointer to the memory region, or nullptr on failure.
+  u8* ReserveMemoryRegion(size_t memory_size);
+
+  // Release a memory region previously reserved with ReserveMemoryRegion().
+  void ReleaseMemoryRegion();
```

Comment:
The use of "<em>a</em> memory region" implies that `ReserveMemoryRegion` can be called multiple times; perhaps it should be "the memory region" (or "the memory region, if any") instead?  (The implementations in the other PRs seem to indicate this is how it should work.)